File: dotnet/Multiplayer/Rooms/Actions/ActionPendingResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Actions
{
    /// <summary>
    /// Response containing pending actions that need to be completed by the host.
    /// Only the host can retrieve pending actions.
    /// </summary>
    /// <typeparam name="T">The type to deserialize action data into.</typeparam>
    public class ActionPendingResponse<T> : ApiResponse<ERoomActionsPendingError> where T : class, new()
    {
        /// <summary>
        /// List of pending actions awaiting completion.
        /// </summary>
        public List<PendingAction<T>> Actions { get; set; } = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/actions/ActionPendingResponse.java
package com.michitai.multiplayer.rooms.actions;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing pending actions that need to be completed by the host.
 * Only the host can retrieve pending actions.
 *
 * @param <T> The type to deserialize action data into.
 */
public class ActionPendingResponse<T> extends ApiResponse {
    @JsonProperty("actions")
    private List<PendingAction<T>> actions;

    public List<PendingAction<T>> getActions() {
        return actions;
    }

    public void setActions(List<PendingAction<T>> actions) {
        this.actions = actions;
    }
}
